manual/README.md enhancements

shindere opened the pull request:
This is a follow-up to a private discussion with @Octachron.

Our initial decision was that @Octachron would shoot first, but I wanted to
keep track of this discussion somehow so I thought it wouldn't hurt
to get the ball rolling.

I am not sure what is the best thing to do though. Since the README
describes where the generated files are written, the commit in this PR
could also be integrated in PR #9997 so that the file does not need to be
modified twice. As you prefer @Octachron.

I am also wondering whether the note about `LD_LIBRARY_PATH` is still
relevant? It feels to me that, if the build system for the manual is not yet
clever enough to include these directories, this is something that
could and should be changed, rather than relying on the user.

gasche commented on manual/README.md:
```diff
@@ -4,7 +4,7 @@ OCAML DOCUMENTATION
 -------------
 
-- Any prerequisites required to build OCaml from sources.
+- Any prerequisites required to build the OCaml compiler itslef from sources.
```
itslef -> itself (in fact I would remove this word completely, "the OCaml compiler" is clear enough)

gasche commented on manual/README.md:
```diff
@@ -15,16 +15,19 @@ Note that you must make sure `hevea.sty` is installed into TeX properly. Your
 NB: If you already set `LD_LIBRARY_PATH` (OS X: `DYLD_LIBRARY_PATH`)
- in your environment don't forget to add
- `otherlibs/unix:otherlibs/str` to it in an absolute way.
+ in your environment don't forget to appen the absolute paths to
```
appen -> append

gasche reviewed (approved):
Good to go once two typos are fixed.

shindere commented:
Thanks alot for the prompt review! The typos are now fixed.

gasche commented:
It looks like #9997 is not ready right now, while this PR is, so I vote to merge it -- when the CI agrees.

shindere commented:
Gabriel Scherer (2021/03/03 11:19 +0000):
> It looks like #9997 is not ready right now, while this PR is, so I
> vote to merge it -- when the CI agrees.

Yes, it makes sensse indeed. Thanks.

Octachron commented on manual/README.md:
```diff
@@ -41,10 +44,11 @@ In the manual:
+The manual is written in an extended dialect of LaTeX and is split in many
+source files. During the build process, these source files are converted into
+classical LaTeX files using the tools available in the `manual/tools`
+directory. These files are then converted to the different output
```
Minor nitpicking: `caml_tex` lives in `tools`.

Octachron commented:
I agree that merging this PR is perfectly fine.

dra27 commented on manual/README.md:
```diff
@@ -15,16 +15,19 @@ Note that you must make sure `hevea.sty` is installed into TeX properly. Your
 
 
-Building
+Building the documentaiton
```
s/documentaiton/documentation

gasche replied:
Actually "manual" would make slightly more sense than "documentation" here, wouldn't it?

dra27 commented on manual/README.md:
```diff
@@ -41,10 +44,11 @@ In the manual:
-source files. During the build process, the sources files are converted into
-classical latex file using the tools available in `tools`. These files are
-then converted to the different output formats using either latex or hevea.
+The manual is written in an extended dialect of LaTeX and is split in many
```
Nit: "split _across_" rather than "split in"

dra27 commented on manual/README.md:
```diff
@@ -62,7 +66,7 @@ chapters (or sometimes sections) are mapped to a distinct `.etex` file:
   - The OCaml language: `refman.etex`
     This chapter consists in a technical description of the OCaml language.
-    Each section of this chapter is mapped to a separated latex file:
+    Each section of this chapter is mapped to a separated LaTeX file:
```
It was there before, but _separate_ rather than _separated_

dra27 reviewed (commented):
1 typo and 2 nits, but it looks like a nice change to me, too

shindere commented:
Florian Angeletti (2021/03/03 03:55 -0800):
> @Octachron commented on this pull request.
> 
> 
> 
> > @@ -41,10 +44,11 @@ In the manual:
>  
>  Source files
>  ------------
> -The manual is written in an extended dialect of latex and is split in many
> -source files. During the build process, the sources files are converted into
> -classical latex file using the tools available in `tools`. These files are
> -then converted to the different output formats using either latex or hevea.
> +The manual is written in an extended dialect of LaTeX and is split in many
> +source files. During the build process, these source files are converted into
> +classical LaTeX files using the tools available in the `manual/tools`
> +directory. These files are then converted to the different output
> 
> Minor nitpicking: `caml_tex` lives in `tools`.

Ah, there seem to be tools in both `manual/tools` and `tools`
directories. Do yout hink the README should mention both of them?

shindere commented:
David Allsopp (2021/03/03 12:20 +0000):
> @dra27 commented on this pull request.
>
> 1 typo and 2 nits, but it looks like a nice change to me, too
>
> > @@ -15,16 +15,19 @@ Note that you must make sure `hevea.sty` is installed into TeX properly. Your
>  package manager may not do this for you. Run `kpsewhich hevea.sty` to check.
>
>
> -Building
> +Building the documentaiton
>
> s/documentaiton/documentation

I replaced documentation by "manual" according to @gasche's suggestion.

> > @@ -41,10 +44,11 @@ In the manual:
>
>  Source files
>  ------------
> -The manual is written in an extended dialect of latex and is split in many
> -source files. During the build process, the sources files are converted into
> -classical latex file using the tools available in `tools`. These files are
> -then converted to the different output formats using either latex or hevea.
> +The manual is written in an extended dialect of LaTeX and is split in many
>
> Nit: "split _across_" rather than "split in"

Fixed. Thanks.

> > @@ -62,7 +66,7 @@ chapters (or sometimes sections) are mapped to a distinct `.etex` file:
>
>    - The OCaml language: `refman.etex`
>      This chapter consists in a technical description of the OCaml language.
> -    Each section of this chapter is mapped to a separated latex file:
> +    Each section of this chapter is mapped to a separated LaTeX file:
>
> It was there before, but _separate_ rather than _separated_

Fixed. Also replaced the other occurrence of "separated" by "divided":
`This part is divided in two very distinct chapters;`.

shindere committed:
manual/README.md enhancements

shindere commented:
Thanks a lot, @Octachron!!!
